From cde4c6ab5e5e45c5ef3bd718245e47ca5c1179c5 Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Sun, 10 Mar 2013 23:41:52 +0200
Subject: [PATCH 13/63] rpc-ub: add m0_net_xprt_init/fini (minor edit)

There is nothing special in this commit, but let #includes be in place.
---
 rpc/ub/README |  8 ++++----
 rpc/ub/ub.c   | 35 ++++++++++++++++++++++++++++++++++-
 2 files changed, 38 insertions(+), 5 deletions(-)

diff --git a/rpc/ub/README b/rpc/ub/README
index 36bcf7f..793ed8b 100644
--- a/rpc/ub/README
+++ b/rpc/ub/README
@@ -13,12 +13,12 @@ of millions.
 
                           +--------------+
      num. of connections  |              |
-    --------------------->|              |   rate (MPS)
-                          |              +---------------->
+    --------------------->|              |  rate (MPS)
+                          |              +--------------->
      num. of slots        |              |
     --------------------->|    rpc-ub    |
-                          |              |   CPU load (%)
-     message length       |              +---------------->
+                          |              |  CPU load (%)
+     message length       |              +--------------->
     --------------------->|              |
                           |              |
                           +--------------+
diff --git a/rpc/ub/ub.c b/rpc/ub/ub.c
index e0e96f0..475d9e9 100644
--- a/rpc/ub/ub.c
+++ b/rpc/ub/ub.c
@@ -18,10 +18,43 @@
  * Original creation date: 04-Mar-2013
  */
 
-#include "lib/ub.h"  /* m0_ub_set */
+#include "lib/ub.h"        /* m0_ub_set */
+#include "net/net.h"       /* m0_net_xprt_init */
+#include "net/bulk_mem.h"  /* m0_net_bulk_mem_xprt */
+
+static struct m0_net_xprt *g_xprt = &m0_net_bulk_mem_xprt;
+
+#if 0 /* XXX <<<<<<< */
+#define NAME(ext) "rpc-ub" ext
+static char *g_argv[] = {
+	NAME(""), "-r", "-p", "-T", "AD", "-D", NAME(".db"),
+	"-S", NAME(".stob"), "-A", NAME(".addb-stob"), "-e", SERVER_ENDPOINT,
+	"-s", "XXX"
+};
+
+static struct m0_rpc_server_ctx g_sctx = {
+	.rsx_xprts            = &g_xprt,
+	.rsx_xprts_nr         = 1,
+	.rsx_argv             = g_argv,
+	.rsx_argc             = ARRAY_SIZE(g_argv),
+	.rsx_service_types    = NULL, /* XXX */
+	.rsx_service_types_nr = 0,    /* XXX */
+	.rsx_log_file_name    = NAME(".log")
+};
+#undef NAME
+
+static struct m0_rpc_client_ctx g_cctx = {
+};
+#endif /* XXX >>>>>>> */
 
 static void XXX(int iter M0_UNUSED)
 {
+	int rc;
+
+	rc = m0_net_xprt_init(g_xprt);
+	M0_UB_ASSERT(rc == 0);
+
+	m0_net_xprt_fini(g_xprt);
 }
 
 struct m0_ub_set m0_rpc_ub = {
-- 
1.8.3.2

